addons/im_livechat/static/src/embed/common/livechat_button.js

```diff
@@ -52,16 +52,42 @@ export class LivechatButton extends Component {
             animateNotification: !(
                 this.livechatService.thread || this.livechatService.shouldRestoreSession
             ),
+            hasAlreadyMovedOnce: false,
+            isButtonShifted: false,
         });
         useMovable({
             cursor: "grabbing",
             ref: this.ref,
             elements: ".o-livechat-LivechatButton",
             onDrop: ({ top, left }) => {
+                if (!this.state.hasAlreadyMovedOnce) {
+                    this.state.hasAlreadyMovedOnce = true;
+                }
                 this.position.left = `${left}px`;
                 this.position.top = `${top}px`;
             },
         });
+        useExternalListener(document.body, "scroll", this._onScroll, { capture: true });
+    }
+
+    _onScroll(ev) {
+        if (!this.ref.el || this.state.hasAlreadyMovedOnce) {
+            return;
+        }
+        const container = ev.target;
+        container.scrollHeight - container.scrollTop === container.clientHeight ?
+            this._shitButton() :
+            this._unshitButton();
+    }
+
+    _shitButton() {
+        this.state.isButtonShifted = true;
```

Comment:
Why do we need this new state variable? It looks unused. Also, I wouldn't have created two dedicated methods only to set `this.position.top`; I'd rather keep those in the `onScroll` handler.

Reply:
yeah absolutly, i don't know why I came up with this idea yesterday. Changing it!